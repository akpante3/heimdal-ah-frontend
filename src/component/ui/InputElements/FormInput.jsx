import React from 'react';

const FormInput = (props = {}) => {
  const {
    errors, name, username, type, classname, placeholder, onChange,
  } = props;
  // console.log(errors);
  return (
    <div className="form-group">
      <input
        name={name}
        value={username}
        type={type}
        className={classname}
        placeholder={placeholder}
        onChange={onChange}
      />
      {errors[name] && errors[name].constructor === Object && (
        <small className="text-danger h5">{errors[name].msg}</small>
      )}
      {errors[name] && errors[name].constructor === Array && (
        <small className="text-danger h5">{errors[name]}</small>
      )}
    </div>
  );
};

export default FormInput;
