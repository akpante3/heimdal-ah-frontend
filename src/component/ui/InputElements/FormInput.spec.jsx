/* eslint-disable no-undef */
import React from 'react';
import FormInput from './FormInput';

describe('The FormInput Component Test Suite', () => {
  it('should render the FormInput component', () => {
    const wrapper = shallow(<FormInput />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <FormInput /> Component', () => {
    const wrapper = shallow(<FormInput />);
    expect(wrapper.length).toBe(1);
    expect(wrapper.find('input')).toBeDefined();
  });

  it('should render exactly one small html tag if an error object and a name is passed into the component', () => {
    const errors = {
      username: ['The username is invalid'],
    };
    const name = 'username';
    const wrapper = shallow(<FormInput errors={errors} name={name} />);

    expect(wrapper.find('small')).toHaveLength(1);
  });

  it('should render exactly one small html tag if an error object and a name is passed into the component', () => {
    const errors = {
      username: { msg: 'The username is invalid' },
    };
    const name = 'username';
    const wrapper = shallow(<FormInput errors={errors} name={name} />);

    expect(wrapper.find('small')).toHaveLength(1);
  });
});
