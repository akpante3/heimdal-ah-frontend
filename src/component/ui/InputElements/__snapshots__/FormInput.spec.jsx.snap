// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The FormInput Component Test Suite should render the FormInput component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          autoComplete="off"
        />,
        undefined,
        undefined,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "off",
          "className": undefined,
          "name": undefined,
          "onChange": undefined,
          "placeholder": undefined,
          "type": undefined,
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            autoComplete="off"
          />,
          undefined,
          undefined,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "off",
            "className": undefined,
            "name": undefined,
            "onChange": undefined,
            "placeholder": undefined,
            "type": undefined,
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
